File: src/test/java/com/github/bgora/rpnlibrary/InputTransformerTest.java
package com.github.bgora.rpnlibrary;

import org.junit.jupiter.api.Test;

import java.util.function.UnaryOperator;

import static org.assertj.core.api.Assertions.assertThat;

public class InputTransformerTest {
    UnaryOperator<String> tested = new InputTransformer(
            new RPNChecker(
                    new DefaultStrategyProvider()));

    @Test
    void shouldReturnEmptyText() {

        String result = tested.apply("");
        assertThat(result).isEmpty();
    }

    @Test
    void shouldReturnTransferredText() {

        String givenEquation = "(1+2)*5";
        String expected = "( 1 + 2 ) * 5";

        String result = tested.apply(givenEquation);

        assertThat(result).isEqualTo(expected);
    }

    @Test
    void shouldReturnTransferredFunctionText() {

        String givenEquation = "sin(15)";
        String expected = "sin ( 15 )";

        String result = tested.apply(givenEquation);

        assertThat(result).isEqualTo(expected);
    }

    @Test
    void shouldReturnTransferredTextForFunctionAndEquation() {

        String givenEquation = "sin(15) + (21 *5)";
        String expected = "sin ( 15 ) + ( 21 * 5 )";

        String result = tested.apply(givenEquation);

        assertThat(result).isEqualTo(expected);
    }
}

File: src/test/kotlin/com/github/bgora/rpnlibrary/InputTransformerTest.kt
package com.github.bgora.rpnlibrary

import io.kotest.core.spec.style.FreeSpec
import io.kotest.matchers.shouldBe

class InputTransformerTest : FreeSpec({
    val tested = InputTransformer(RPNChecker(DefaultStrategyProvider()))
     "should Return empty Text" - {
         val text = ""
         val result = tested.apply(text)
         result shouldBe ""
                                          }
    "should Return transferred Text" - {
        val text = "(1+2)*5"
        val result = tested.apply(text)
        result shouldBe "( 1 + 2 ) * 5"
    }
     "should Return Transferred Function Text" -{
         val text = "sin(15)"
         val result = tested.apply(text)
         result shouldBe "sin ( 15 )"

     }

     "should Return Transferred Text For Function and equation" -{
        val text = "sin(15) + (21 *5)"
        val result = tested.apply(text)
        result shouldBe "sin ( 15 ) + ( 21 * 5 )"

     }
 })
